horace_core/sqw/PixelData/@PixelData/get_data.m

```diff
@@ -0,0 +1,130 @@
+function data_out = get_data(obj, pix_fields, varargin)
+% Retrive data for a field, or fields, for the given pixel indices in
+% the current page. If no pixel indices are given, all pixels in the
+% current page are returned.
+%
+% This method provides a convinient way of retrieving multiple fields
+% of data from the pixel block. When retrieving multiple fields, the
+% columns of data will be ordered corresponding to the order the fields
+% appear in the inputted cell array.
+%
+%   >> sig_and_err = pix.get_data({'signal', 'variance'})
+%        retrives the signal and variance over the whole range of pixels
+%
+%   >> run_det_id_range = pix.get_data({'run_idx', 'detector_idx'}, 4:10);
+%        retrives the run and detector IDs for pixels 4 to 10
+%
+% Input:
+% ------
+%   pix_fields       The name of a field, or a cell array of field names
+%   abs_pix_indices  The pixel indices to retrieve, if not given, get full range
+%
+[pix_fields, abs_pix_indices] = parse_args(obj, pix_fields, varargin{:});
+field_indices = cell2mat(obj.FIELD_INDEX_MAP_.values(pix_fields));
+
+if obj.is_file_backed_()
+
+    base_pg_size = obj.max_page_size_;
+    if abs_pix_indices == -1
```

Comment:
you're using `-1` as a magic number across this file for "no pixel indexes". A named constant would be clearer.